icechunk-python/src/storage.rs

```diff
@@ -33,6 +30,22 @@ impl From<&PyS3Credentials> for StaticS3Credentials {
     }
 }
 
+impl From<PyS3Credentials> for StaticS3Credentials {
+    fn from(credentials: PyS3Credentials) -> Self {
+        StaticS3Credentials {
+            access_key_id: credentials.access_key_id,
+            secret_access_key: credentials.secret_access_key,
+            session_token: credentials.session_token,
+        }
+    }
+}
```

Comment:
It's useful to have a value-to-value `From` impl instead of just an impl on `&PyS3Credentials` so that you can avoid the clones when not needed.